### DISCUSSION TITLE
Where's the issue in my theme?

### DISCUSSION THREAD

**Discussion by DIrgi** [mapsforge/vtm#1121] [Q&A] [answered]
Hi,
I have issues with my theme code and I don't know where the problem is.
Target is, displaying all tracks. belonging to highway, wich are allowed for motor vehicles, classified by grades.

This is what I see, when I check the object:
![image](https://github.com/mapsforge/vtm/assets/57968526/39ac9da1-2d2b-430f-a62d-47571b625a8f)

When I use the following code:
```
<m e="way">
	<m k="highway" v="track">
		<m k="motor_vehicle" v="yes">
			<m k="tracktype" v="grade1">
				<line use="track_grade1" />
				<text use="highway-track" />
			</m>
			<m k="tracktype" v="grade2>
				<line use="track_grade2" />
				<text use="highway-track" />
			</m> 
                        .....
                </m>
      </m>
</m>
```
I see absolutely nothing. When I request without "motor_vehicle" like this:

```
<m e="way">
	<m k="highway" v="track">
	      <m k="tracktype" v="grade1">
		      <line use="track_grade1" />
	              <text use="highway-track" />
	      </m>
             -----
             -----
```

then I see every (graded) track (this includes all tracks independent of "motor_vehicle" what I don't want to have)
Where the heck is the problem?
Can someone help me please?

Thank you

**Comment by devemux86** [comment, marked as answer]
The tags that can be used in map themes must be included first in the map files.

The tags that are included in the map files are declared in the [tag-mapping.xml](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml).
e.g. track_type exists, motor_vehicle no

**Reply by DIrgi** [reply]
Ahhhh, I seee - so motor_vehicle is not included.....this declares some things.....
Is it possible to have this declaration in the next mapforge map updates?

because....with this definition it clearly mark a track for vehicle offroad usage.
Without well. every track is being displayed even it's not usable or forbidden and 
unfortunately a grade level is not an indicator for permission.

**Reply by devemux86** [reply]
@DIrgi If you can collect what extra tags are needed, we can include them.

The tags must be declared with the necessary values for the map themes.

e.g. see the [motor_vehicle](https://wiki.openstreetmap.org/wiki/Key:motor_vehicle) in OpenStreetMap Wiki

**Reply by DIrgi** [reply]
That's great.
I checked the tag-info source.
It has more values then I though (hundreds) however quite often tagging issues it think.
Never the less this is the list with ALL tagged values - descending based on their amount (count)

Finally I would go only for:
![image](https://github.com/mapsforge/vtm/assets/57968526/9ffa6112-dc1f-4733-9449-28821e83535b)

Cleaned up, it's just this: (maximum)
![image](https://github.com/mapsforge/vtm/assets/57968526/d3f27891-1637-42a0-bc17-493abfe65e5e)

And here for easy copy/paste, just the values:
no,yes,private,destination,agricultural,forestry,designated,permissive,permit,customers,delivery,unknown,bus,emergency,official,service,mining,atv

But, some of the lower ones we still can skip - I don't care. Probably everything downwards from bus.

**Reply by devemux86** [reply]
@DIrgi thanks for searching it.

Don't we just need in the themes a 2 value option, motor_vehicle yes / no?
It can have equivalent values, e.g. see the [oneway declaration](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-map-writer/src/main/config/tag-mapping.xml#L640-L642).

**Reply by DIrgi** [reply]
I'm fine with just yes/no
I understand your question to add even some more.

**Reply by devemux86** [reply]
@DIrgi So we can start by adding the yes / no combinations.

Are there equivalent yes / no values in the OSM database, like with the oneway tag?

**Reply by DIrgi** [reply]
No, I scanned the complete db - Just yes/no

**Reply by DIrgi** [reply]
when will be your next update run on the map?

**Reply by DIrgi** [reply]
although it's not directly related but you as an Expert...before I got lost in the search again...
Is it possible to replace something?

i.e.   v="test"  replace by "new" ? and if? - How?
Thanks in advance

**Reply by devemux86** [reply]
> when will be your next update run on the map?

Rebuilding all maps usually takes some time.

> i.e. v="test" replace by "new" ? and if? - How?

What do you mean, use different values from the database?
The rules in themes use the values existing in the maps that come from the OSM database.

**Reply by DIrgi** [reply]
No worries, I just wanted to know when is normally your plan to renew the mapforges maps?
quarterly, half year, once a year - and when is your next run?

Replacement. OK - An example.
Let's say a track has a tracktype:-> "grade1"
When I tell the system please use tracktype as name to name a track (in the map)  it will be named "grade1".
I want to know if I can replace the tracktype name "grade1" with i.e "g1" before I name (print out) the track in the map?

Like:
If tracktype is "grade1", replace the name by "g1" and show the track name as "g1" (and not as "grade1") in the map.

**Reply by devemux86** [reply]
@DIrgi Do you mean customize the values to also use them as names?

Poi names come from the name / addr:housenumber / ele fields.
Way names come from the name / addr:housenumber / ref fields.

I don't remember if the tag-transform rule can be used for this.
(e.g. see the [default theme](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/default.xml) at the top)

**Reply by DIrgi** [reply]
that was the keyword "tag-transform" I'll try, thank you

**Comment by devemux86** [comment]
I added the motor_vehicle in the tag-mapping, it will be included in the next map files:
- https://github.com/mapsforge/mapsforge/pull/1481

**Reply by DIrgi** [reply]
Thank you very much
